camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/common/KafkaReaderTest.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.camus.etl.kafka.common;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.URI;
+
+import kafka.javaapi.FetchRequest;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.consumer.SimpleConsumer;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.TaskAttemptID;
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.linkedin.camus.etl.kafka.CamusJob;
+
+
+public class KafkaReaderTest {
+  private KafkaReader kafkaReader;
+
+  @Before
+  public void before() throws Exception {
+    Configuration conf = new Configuration();
+    conf.set(CamusJob.KAFKA_CLIENT_NAME, "DummyClientName");
+    TaskAttemptContext context = new TaskAttemptContext(conf, new TaskAttemptID());
+    EtlRequest request = new EtlRequest();
+    request.setOffset(0);
+    request.setLatestOffset(1);
+    request.setURI(new URI("http://localhost:8888"));
+    this.kafkaReader = new KafkaReader(context, request, 100, 100);
+  }
+
+  @Test
+  public void testFetchFailure() throws SecurityException, NoSuchFieldException,
+    IllegalArgumentException, IllegalAccessException, IOException {
+    SimpleConsumer consumer = EasyMock.createNiceMock(SimpleConsumer.class);
+    EasyMock.expect(consumer.fetch((FetchRequest) EasyMock.anyObject())).andReturn(null);
+    EasyMock.expectLastCall().times(2);
```

Comment:
One for the first attempt, and one for the retry, after refreshing topic metadata.